apps/native/app/src/ui/lib/progress-meter/progress-meter.tsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react'
+import { View } from 'react-native'
+import { screenWidth } from '../../../utils/dimensions'
+import styled, { useTheme } from 'styled-components/native'
+import { Typography } from '../typography/typography'
+
+const Host = styled.View`
+  border-radius: ${({ theme: { border } }) => border.radius.extraLarge};
+  height: 16px;
+  padding: ${({ theme }) => theme.spacing.smallGutter}px;
+  margin-top: ${({ theme }) => theme.spacing[1]}px;
+`
+
+const InnerHost = styled.View`
+  flex: 1;
+  flex-direction: row;
+  column-gap: ${({ theme }) => theme.spacing.smallGutter}px;
+  border-radius: ${({ theme: { border } }) => border.radius.large};
+  height: 100%;
+`
+
+export type ProgressMeterVariant = 'blue' | 'red' | 'rose' | 'mint'
+
+interface ProgressMeterProps {
+  draftTotalSteps: number
+  draftFinishedSteps: number
+  containerWidth?: number
+  variant?: ProgressMeterVariant
+  progressMessage?: string
+}
+
+const colorSchemes = {
+  blue: {
+    unfinished: 'blue200',
+    outer: 'blue100',
+    inner: 'blue400',
+  },
+  red: {
+    unfinished: 'red200',
+    outer: 'red100',
+    inner: 'red400',
+  },
+  rose: {
+    unfinished: 'roseTinted200',
+    outer: 'roseTinted100',
+    inner: 'roseTinted400',
+  },
+  mint: {
+    unfinished: 'mint200',
+    outer: 'mint100',
+    inner: 'mint600',
+  },
+} as const
+
+export const ProgressMeter = ({
+  draftFinishedSteps,
+  draftTotalSteps,
+  progressMessage,
+  containerWidth,
+  variant = 'blue',
+}: ProgressMeterProps) => {
+  const theme = useTheme()
+  const steps = Array.from(Array(draftTotalSteps ?? 1).keys())
+  const allowedWidth = containerWidth ?? screenWidth - theme.spacing[2] * 4
+
+  if (!draftTotalSteps) {
```

Comment:
Is this to check for zero? Be cause the prop is required

Reply:
the prop from the api is optional so I send in `progressTotalSteps ?? 0`. But I could also make it optional and then it will be undefined, ends in the same result 😊